judge/contest_format/ioi16.py

```diff
@@ -0,0 +1,95 @@
+                                  FROM judge_contestproblem cp
+                                           INNER JOIN
+                                       judge_contestsubmission cs
+                                       ON (cs.problem_id = cp.id AND cs.participation_id = %s)
+                                           LEFT OUTER JOIN
+                                       judge_submission sub ON (sub.id = cs.submission_id)
+                                           INNER JOIN judge_submissiontestcase tc
+                                                      ON sub.id = tc.submission_id
+                                  GROUP BY cp.id, tc.batch, sub.id
+                              ) q
+                                  INNER JOIN (
+                             SELECT prob, batch, MAX(r.batch_points) as max_batch_points
+                             FROM (
+                                      SELECT cp.id          as `prob`,
+                                             tc.batch       as `batch`,
+                                             MIN(tc.points) as `batch_points`
+                                      FROM judge_contestproblem cp
+                                               INNER JOIN
+                                           judge_contestsubmission cs
+                                           ON (cs.problem_id = cp.id AND cs.participation_id = %s)
+                                               INNER JOIN
+                                           judge_submission sub ON (sub.id = cs.submission_id)
+                                               INNER JOIN judge_submissiontestcase tc
+                                                          ON sub.id = tc.submission_id
+                                      GROUP BY cp.id, tc.batch, sub.id
+                                  ) r
+                             GROUP BY prob, batch
+                         ) p
+                                       ON p.prob = q.prob AND p.batch = q.batch
+                         WHERE p.max_batch_points = q.batch_points
+                           AND p.prob = q.prob
+                         GROUP BY q.prob, q.batch
+                     ) best
+            ''', (participation.id, participation.id))
+
+            for problem_id, time, subtask_points in cursor.fetchall():
+                time = from_database_time(time)
+                if self.config['cumtime']:
+                    dt = (time - participation.start).total_seconds()
+                else:
+                    dt = 0
+
+                if format_data.get(str(problem_id)) is None:
+                    format_data[str(problem_id)] = {'points': 0, 'time': 0}
+                format_data[str(problem_id)]['points'] += subtask_points
+                format_data[str(problem_id)]['time'] = max(dt, format_data[str(problem_id)]['time'])
+
+            for problem_id in format_data:
+                penalty = format_data[problem_id]['time']
```

Comment:
```suggestion
                penalty = problem_data['time']
```

(same for the following line)